Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -177,17 +158,17 @@ open class ChartDataSet: ChartBaseDataSet
     open override var xMax: Double { return _xMax }
     
     /// - returns: The number of y-values this DataSet represents
-    open override var entryCount: Int { return _values?.count ?? 0 }
+    open override var entryCount: Int { return values.count }
     
     /// - returns: The entry object found at the given index (not x-value!)
     /// - throws: out of bounds
     /// if `i` is out of bounds, it may throw an out-of-bounds exception
     open override func entryForIndex(_ i: Int) -> ChartDataEntry?
     {
-        guard i >= 0 && i < _values.count else {
+        guard values.indices.contains(i) else {
```

Comment:
seems this is O(n) cost, while the old one is O(1)?

Reply:
Fixed

Reply:
@liuxuan30 For future reference, I've looked into this, and all `Range` types have O(1) complexity. They perform a bounds check internally instead of iterating through the range. So this:
```
(values.startIndex..<values.endIndex).contains(i)
```
Performs the same, and, I feel, is more clear.